lib/SILGen/SILGenConstructor.cpp

```diff
@@ -776,17 +791,47 @@ void SILGenFunction::emitClassConstructorInitializer(ConstructorDecl *ctor) {
   if (selfClassDecl->isRootDefaultActor() && !isDelegating) {
     SILLocation PrologueLoc(selfDecl);
     PrologueLoc.markAsPrologue();
+    emitDefaultActorInitialization(*this, PrologueLoc, selfArg);
+  }
 
-    if (selfClassDecl->isDistributedActor() &&
-        ctor->isDistributedActorResolveInit()) {
-      auto addressArg  = false; // TODO: get the address argument
-      auto transportArg  = false; // TODO: get the transport argument
-      emitDistributedRemoteActorInitialization(*this, PrologueLoc,
-          selfArg, addressArg, transportArg);
-    } else {
-      // is normal (default) actor
-      emitDefaultActorInitialization(*this, PrologueLoc, selfArg);
-    }
+  // Initialize the distributed actor identity and transport.
+  if (selfClassDecl->isDistributedActor()) {
+    if (ctor->isDistributedActorLocalInit()) {
+      SILLocation PrologueLoc(selfDecl);
+      PrologueLoc.markAsPrologue();
+
+      // TODO(distributed): actually this has to be emitted into every designated distributed
+      //       actor init, not just the "local init one"; it becomes just a
+      //       special case that replaces the no arg initializer for distributed actors
```

Comment:
I think this work should be in a separate PR, with this PR focused on just providing the implementation of DistributedActorStorage as part of the layout of every distributed actor instance, plus built-ins to access its fields. Calling a different built-in to initialize the DistributedActorStorage of a class is kind of a separate concern, imo.

Reply:
Yeah I'm happy to split things into many PRs -- right now I wanted to share this to ask for help on what I'm stuck with, so the "messy everything I have" PR is just this